test/Writer/Factory/WriterFactoryTest.php

```diff
@@ -0,0 +1,290 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace ZendTest\Log\Writer\Factory;
+
+use Interop\Container\ContainerInterface;
+use PHPUnit_Framework_TestCase as TestCase;
+use Zend\Log\Writer\Factory\WriterFactory;
+use Zend\ServiceManager\AbstractPluginManager;
+use Zend\ServiceManager\ServiceLocatorInterface;
+use ZendTest\Log\Writer\TestAsset\InvokableObject;
+
+class WriterFactoryTest extends TestCase
+{
+    /**
+     * @covers \Zend\Log\Writer\Factory\WriterFactory::setCreationOptions
+     */
+    public function testSetCreationOptions()
+    {
+        // Arrange
+        $container = $this->getMockForAbstractClass(ServiceLocatorInterface::class);
```

Comment:
You're mixing PHPUnit mocks with Prophecy mocks; use one or the other. This one can become:

```php
$container = $this->prophesize(ServiceLocatorInterface::class);
$container->willImplement(ContainerInterface::class);
$container = $container->reveal();
```